x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/package-info.java

```diff
@@ -0,0 +1,158 @@
+ *         should take only a few seconds. This is a fast path to running ESQL's integration tests.
+ *     </li>
+ *     <li>
+ *         Pick one of the csv-spec files in {@code x-pack/plugin/esql/qa/testFixtures/src/main/resources/}
+ *         and add a test for the function you want to write. These files are roughly themed but there
+ *         isn't a strong guiding principle in the theme.
+ *     </li>
+ *     <li>
+ *         Rerun the {@code CsvTests} and watch your new test fail. Yay, TDD doing it's job.
+ *     </li>
+ *     <li>
+ *         Find a function in this package similar to the one you are working on and copy it to build
+ *         yours. There's some ceremony required in each function class to make it constant foldable
+ *         and return the right types. Take a stab at these, but don't worry too much about getting
+ *         it right.
+ *     </li>
+ *     <li>
+ *         There are also methods annotated with {@link org.elasticsearch.compute.ann.Evaluator}
+ *         that contain the actual inner implementation of the function. Modify those to look right
+ *         and click {@code Build->Recompile 'FunctionName.java'} in IntelliJ. This should generate
+ *         an {@link org.elasticsearch.compute.operator.EvalOperator.ExpressionEvaluator} implementation
+ *         calling the method annotated with {@link org.elasticsearch.compute.ann.Evaluator}.
+ *     <li>
+ *         Once your evaluator is generated you can implement {@link org.elasticsearch.xpack.esql.planner.Mappable#toEvaluator},
+ *         having it return the generated evaluator.
+ *     </li>
+ *     <li>
+ *         Add your function to {@link org.elasticsearch.xpack.esql.expression.function.EsqlFunctionRegistry}.
+ *         This links it into the language and {@code SHOW FUNCTIONS}. Also add your function to
+ *         {@link org.elasticsearch.xpack.esql.io.stream.PlanNamedTypes}. This makes your function
+ *         serializable over the wire. Mostly you can copy existing implementations for both.
+ *     </li>
+ *     <li>
+ *         Rerun the {@code CsvTests}. They should find your function and maybe even pass. Add a
+ *         few more tests in the csv-spec tests. They run quickly so it isn't a big deal having
+ *         half a dozen of them per function. In fact, it's useful to add more complex combinations
+ *         of things here, just to catch any accidental strange interactions. For example, it is
+ *         probably a good idea to have your function passes as a parameter to another function
+ *         like {@code EVAL foo=MOST(0, MY_FUNCTION(emp_no))}. And likely useful to try the reverse
+ *         like {@code EVAL foo=MY_FUNCTION(MOST(languages + 10000, emp_no)}.
+ *     </li>
+ *     <li>
+ *         Now it's time to make a unit test! The infrastructure for these is under some flux at
+ *         the moment, but it's good to extend from {@code AbstractScalarFunctionTestCase}. All of
```

Comment:
```suggestion
 *         Now it's time to make a unit test! The infrastructure for these might change, but it's good to extend from {@code AbstractScalarFunctionTestCase}. All of
```

While admittedly everything about this guide might change, it's likely we'll forget to update this sentence short term and this will remain in flux for a longer while.